core/modules/menu/js/menu.admin.js

```diff
@@ -19,6 +19,60 @@ Backdrop.behaviors.menuAdminFieldsetSummaries = {
   }
 };
 
+Backdrop.behaviors.menu_edit_item_parents = {
+  attach: function (context, settings) {
+    // Get the menu parent options from settings.
+    var menuOptions = settings.menu_edit_item_parents;
+    // Get the current menu name from settings.
+    var menuName = settings.menu_edit_item_menu;
+    // Always move the menu selector before the menu parent selector.
+    $('.form-item-parent-menu').insertBefore('#menu-parent-select-wrapper');
```

Comment:
Is this necessary? Could this be done by setting `#weight`?

Reply:
This whole thing of using JS is because contrib tends to alter the form structure and move things around and into fieldsets and such. Thus weights could change, and the two selects become separated.